### ISSUE TITLE
Some questions about the implementation

### ISSUE THREAD

**Issue by kan-bayashi** [NVIDIA/tacotron2#25]
Hi, thank you for sharing the great work.
I have some questions about the implementation.

First question is about padding in data preparation.
https://github.com/NVIDIA/tacotron2/blob/master/data_utils.py#L103-107
The value of `gated_padded[i]` will be all of 0 when the i-th length is the maximum length.
Is it correct?

Second question is about network architecture of decoder.
https://github.com/NVIDIA/tacotron2/blob/master/model.py#L341
The lstm structure looks like different from the original.
Is there any specific reason?

Thank you in advance!

**Comment by rafaelvalle** [comment]
Hey @kan-bayashi.
1) It not really correct. Note that gate_padded is of length maximum mel length + 1. When updating gate_padded, we use the non-padded mels whose maximum length is maximum mel length.

2) What difference are you referring to? I think you refer to the 2 LSTMs on the decoder but want to make sure before replying.

**Comment by kan-bayashi** [comment]
Hi @rafaelvalle. Thank you for quick response!

1. 
Oh I overlooked the size of padded tensor. 
Now I understood. Thank you.

2. 
Yes. I refer to 2 LSTM part. 
In my understanding, the inputs of first lstm layer is prenet outputs + encoder embedding, and the inputs of second layer is outputs of first lstm layer.
Finally, outputs of second layer is concatenated with encoder embedding, and those are inputted into projection layer.
But in the case of your implementation, it looks like different.

**Comment by rafaelvalle** [comment]
2) We are currently testing the performance of 3 different implementations. The one that seems to resemble the paper the most, i.e. 2 Decoder LSTMs and 1 Attention LSTM is the slowest to learn attention but reaches lower training and validation losses much quicker. 
Having said that, we need to run more tests to evaluate how important the difference in loss is.
We're also evaluating adding zoneout with the hope that it will decrease the time it takes to learns attention.

**Comment by kan-bayashi** [comment]
@rafaelvalle Thank you for your answer. I closed this issue.
